src/mainwindow.cpp

```diff
@@ -697,6 +700,37 @@ void MainWindow::find()
     consoleTabulator->terminalHolder()->currentTerminal()->impl()->toggleShowSearchBar();
 }
 
+void MainWindow::sendHistoryToEditor()
+{
+    const QString fn = QStandardPaths::writableLocation(QStandardPaths::TempLocation) + QLatin1String("/qterminal.history.") + QString::number(QCoreApplication::applicationPid());
+    QFile file(fn);
+    if (!file.open(QIODevice::WriteOnly)) {
+        qDebug() << "Failed to open" << file.fileName() << "for writing";
+        return;
+    }
+    TermWidgetImpl *impl = consoleTabulator->terminalHolder()->currentTerminal()->impl();
+    impl->saveHistory(&file);
+    file.close();
+    Properties::EditorType type = Properties::Instance()->editorType;
+    QString editor = Properties::Instance()->editor;
+    if (type == Properties::Editor_Auto) {
+        type = Properties::Editor_GUI;
+        for (const char *ed : { "vim", "vi", "less", "more", "nano", "pico", "view", "ed" }) {
+            if (editor == QLatin1String(ed)) {
+                type = Properties::Editor_Terminal;
+                break;
+            }
+        }
+    }
+    if (type == Properties::Editor_Terminal) {
+        impl->sendText(QLatin1String("\1#\n")); // comment out current line first
+        impl->sendText(Properties::Instance()->editor + QLatin1Char(' ') + fn + QLatin1Char('\n'));
```

Comment:
Currently it's unlikely that filenames contain spaces or quotes. However, if we place the history file to somewhere else, the possibility will be higher. Such cases should be handled.

Reply:
Also, the current process may not be a shell (e.g., `qterminal -e python`). I think it's possible to query the process owning the current PTY - lots of platform-dependent code needed, though.